sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py

```diff
@@ -230,12 +242,21 @@ def _to_rest_object(self) -> ComputeResource:
                 )
             )
 
+        idle_time_before_shutdown = None
+        if self.idle_time_before_shutdown_minutes:
+            idle_time_before_shutdown = f"PT{self.idle_time_before_shutdown_minutes}M"
+        elif self.idle_time_before_shutdown:
+            module_logger.warning(
+                "Property 'idle_time_before_shutdown': This property is deprecated. Please use 'idle_time_before_shutdown_minutes' instead."
+            )
+            idle_time_before_shutdown = self.idle_time_before_shutdown
```

Comment:
If you wanted to be extra cautious - you could also check for the scenario where _both_ of these values were supplied and raise some kind of error.

Reply:
I believe it would be best to leave it without an error. Since _load_from_rest() sets the idle_time_before_shutdown property, throwing an error here would mean that a compute instance returned from the API couldn't be used to create a copy of the same instance.